extension/js/common/core/crypto/pgp/msg-util.ts

```diff
@@ -148,80 +143,33 @@ export class MsgUtil {
-        if (!verifyRes.signer) {
-          // todo - currently only the first signer will be reported. Should we be showing all signers? How common is that?
-          verifyRes.signer = {
-            longid: OpenPGPKey.bytesToLongid(verification.keyid.bytes),
-            primaryUserId: await OpenPGPKey.getPrimaryUserId(pubs, verification.keyid)
-          };
-        }
-      }
-    } catch (verifyErr) {
-      verifyRes.match = null; // tslint:disable-line:no-null-keyword
-      if (verifyErr instanceof Error && verifyErr.message === 'Can only verify message with one literal data packet.') {
-        verifyRes.error = 'FlowCrypt is not equipped to verify this message';
-        verifyRes.isErrFatal = true; // don't try to re-fetch the message from API
-      } else if (verifyErr instanceof Error && verifyErr.message.startsWith('Insecure message hash algorithm:')) {
-        verifyRes.error = `Could not verify message: ${verifyErr.message}. Sender is using old, insecure OpenPGP software.`;
-        verifyRes.isErrFatal = true; // don't try to re-fetch the message from API
-      } else if (verifyErr instanceof Error && verifyErr.message === 'Signature is expired') {
-        verifyRes.error = verifyErr.message;
-        verifyRes.isErrFatal = true; // don't try to re-fetch the message from API
-      } else if (verifyErr instanceof Error && verifyErr.message === 'Message digest did not match') {
-        verifyRes.error = verifyErr.message;
-      } else {
-        verifyRes.error = `Error verifying this message: ${String(verifyErr)}`;
-        Catch.reportErr(verifyErr);
-      }
-    }
-    return verifyRes;
-  };
-
-  public static verifyDetached: PgpMsgMethod.VerifyDetached = async ({ plaintext, sigText }) => {
+  public static verifyDetached: PgpMsgMethod.VerifyDetached = async ({ plaintext, sigText, verificationPubs }) => {
     const message = opgp.message.fromText(Buf.fromUint8(plaintext).toUtfStr());
     await message.appendSignature(Buf.fromUint8(sigText).toUtfStr());
-    const keys = await MsgUtil.getSortedKeys([], message);
-    return await MsgUtil.verify(message, keys.forVerification, keys.verificationContacts[0]);
+    return await OpenPGPKey.verify(message, verificationPubs);
   };
 
-  public static decryptMessage: PgpMsgMethod.Decrypt = async ({ kisWithPp, encryptedData, msgPwd }) => {
+  public static decryptMessage: PgpMsgMethod.Decrypt = async ({ kisWithPp, encryptedData, msgPwd, verificationPubs }) => {
     const longids: DecryptError$longids = { message: [], matching: [], chosen: [], needPassphrase: [] };
     let prepared: PreparedForDecrypt;
     try {
       prepared = await PgpArmor.cryptoMsgPrepareForDecrypt(encryptedData);
     } catch (formatErr) {
       return { success: false, error: { type: DecryptErrTypes.format, message: String(formatErr) }, longids };
     }
+    if (prepared.isCleartext) {
+      // todo: error if no verificationPubs?
```

Comment:
But what if this is a signed-only pkcs7 message, wouldn't it go into this OpenPGP branch and fail? Mabe need to add another if or another condition in this if so that we handle the issue gracefully.

Reply:
`isCleartext` and `isPkcs7` are mutually exclusive, so it won't go to this branch

Reply:
Ah. no such thing as a signed-only pkcs7 message?

Reply:
Added appropriate remarks

Reply:
Please see the updated comment